File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
using namespace std;
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> s;
        for (int i=0;i<size(nums);i++){
            s.insert(nums[i]);
        }
        size_t len=s.size();
        if(size(nums)== int(len)){
            return false;
        }else{
            return true;
        }    
    }
};

File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result =[]
        nums.sort()
        for i in range(len(nums)- 2):
            target = -nums[i]
            right = i+1
            left = len(nums) - 1
            while(right < left ):
                sum = nums[ right ] + nums[left]
                if(sum > target):
                    left -= 1
                elif( sum < target ):
                    right += 1
                else :
                    result.append([nums[i],nums[right] , nums[left]])
                    right += 1
                    left -= 1
        
        r = [list(x) for x in dict.fromkeys(map(tuple, result))]
        return r

                
            

        